### PR TITLE
Native AmigaOS4 Makefile

### LINKED ISSUE #24: Native makefile
Native, simple, makefile would be nice to have.

Using configure + make has libtool-related (?) issue which prevents shared object building unless makefile is modified (remove -no-undefined).

Cmake works natively, but cmake as a tool is not very well supported on the native platform.

### PR DESCRIPTION
**Pull request by BSzili**
It can be used to build the static and shared library, and install them. The SDL_config_amigaos4.h header might need some tweaks, because I don't have the GLES2 headers. This should fix #24.

### CODE REVIEW

**Comment by capehill** [comment]
Nice, this should be very handy. Some things that might be useful:

a) optimization flag 
b) ggdb flag to provide debug symbols
c) optional altivec: apparently it's not possible to provide a lib that support both altivec and non-altivec machines
d) render/ paths (or did I miss them?)

**Comment by kas1e** [comment]
@Capehil
I remember that one of brothers (Thomas?), said that the only debug flag which 100% works fine and can be used on amiga are -gstabs. Other ones may works, but sometime may not

**Comment by BSzili** [comment]
> a) optimization flag
> b) ggdb flag to provide debug symbols

How did I miss the optimization and debug flags completely? I'll add these soon. 
> c) optional altivec: apparently it's not possible to provide a lib that support both altivec and non-altivec machines

AltiVec it's a bit tricky, because without -maltivec the intrinsics didn't work at all, so I'll probably disable it completely for the time being.

> d) render/ paths (or did I miss them?)

Could you clarify what you mean?

**Comment by afxgroup** [comment]
use gstabs instead of ggdb when generating symbols. because if i remember correctly there was problems reading them with addr2line on os4

**Comment by capehill** [comment]
> 
> > d) render/ paths (or did I miss them?)
> 
> Could you clarify what you mean?

This directory should also be built: https://github.com/AmigaPorts/SDL/tree/master/src/render

If I looked correctly, it was missing in the makefile.

**Comment by BSzili** [comment]
I got it. I added two commits, let me know if there's anything else I should change.

**Comment by kas1e** [comment]
@BSzili
Should't be also src/render/opengles2/ be in makefile as well ? Currently it builded by default inside of SDL2.

**Comment by BSzili** [comment]
Beats me. I can't build it, because I don't have the SDK, but someone else can add it to the Makefile and config header.

**Comment by kas1e** [comment]
@BSZili
Just in case you will ever need it, there is archive with latest ogles2 sdk:
http://kas1e.mikendezign.com/aos4/ogles2/include_ogles2_2019_02_20.zip

**Comment by BSzili** [comment]
Thanks, but is it OK to distribute it publicly? Either way I'll try if I can build the GLES2 support/renderer.

File: test/Makefile.amigaos4 [hunk 1]
```diff
@@ -0,0 +1,282 @@
+# Makefile to build the AmigaOS4 SDL tests
+
+CC		= ppc-amigaos-gcc
+CFLAGS	= -g -O2 -Wall -g -I../include
+LIBS	= -L.. -lSDL2 -lpthread -lGL -lm
```
Comment (capehill) [inline test/Makefile.amigaos4]: I think -lGL and -lm are unnecessary: or did you have some issues without?

File: Makefile.amigaos4 [hunk 2]
```diff
@@ -13,20 +13,25 @@ TARGET_SHARED  = libSDL2-2.0.so
...
 	./src/file/*.c \
+	./src/filesystem/amigaos4/*.c \
 	./src/haptic/*.c \
 	./src/haptic/dummy/*.c \
 	./src/joystick/*.c \
 	./src/joystick/amigaos4/*.c \
 	./src/loadso/amigaos4/*.c \
 	./src/render/*.c \
+	./src/power/*.c \
+	./src/power/*.c \
```
Comment (capehill) [inline Makefile.amigaos4]: Line 31 seems extra

File: test/Makefile.amigaos4 (hunk 1 above)
Comment (BSzili) [reply test/Makefile.amigaos4]: Some examples need them. Should I link them on a case by case basis?

File: Makefile.amigaos4 (hunk 2 above)
Comment (BSzili) [reply Makefile.amigaos4]: I didn't catch this, I'll remove the extra line.

File: test/Makefile.amigaos4 (hunk 1 above)
Comment (BSzili) [reply test/Makefile.amigaos4]: Never mind, the tests built just fine without them. This was probably a remnant of me trying to build them for Windows and forgot about it.

**Pull request merged**